pkg/tfbridge/detailed_diff.go

```diff
@@ -497,7 +530,23 @@ func MakeDetailedDiffV2(
 	props = stripSecretsAndOutputs(props)
 	newInputs = stripSecretsAndOutputs(newInputs)
 	differ := detailedDiffer{ctx: ctx, tfs: tfs, ps: ps, newInputs: newInputs}
-	return differ.makeDetailedDiffPropertyMap(priorProps, props)
+	res := differ.makeDetailedDiffPropertyMap(priorProps, props)
+
+	if replaceOverride != nil {
+		if containsReplace(res) && !*replaceOverride {
+			for k, v := range res {
+				res[k] = demoteToNoReplace(v)
+			}
+		}
+
+		if !containsReplace(res) && *replaceOverride {
+			// We use the internal __meta property to trigger a replace when we have failed to
+			// determine the correct detailed diff for it.
+			res["__meta"] = &pulumirpc.PropertyDiff{Kind: pulumirpc.PropertyDiff_UPDATE_REPLACE}
+		}
```

Comment:
nit: The logic makes sense. I think this ordering reads much easier.
```suggestion
		if *replaceOverride {
			// We need to make sure there is a replace.
			if !containsReplace(res) {
				// We use the internal __meta property to trigger a replace when we have failed to
				// determine the correct detailed diff for it.
				res["__meta"] = &pulumirpc.PropertyDiff{Kind: pulumirpc.PropertyDiff_UPDATE_REPLACE}
			}
		} else {
			// There is an override for no replaces, so ensure we don't have any.
			for k, v := range res {
				res[k] = demoteToNoReplace(v)
			}
		}
```